server/neptune/workflows/internal/deploy/revision/queue/queue.go

```diff
@@ -54,6 +67,21 @@ func (q *Deploy) Pop() (terraform.DeploymentInfo, error) {
 	return q.queue.Pop()
 }
 
+func (q *Deploy) Contains(msg terraform.DeploymentInfo) bool {
+	if msg.Root.Trigger == activity.ManualTrigger {
+		return q.queue.Contains(msg, High)
+	}
+	return q.queue.Contains(msg, Low)
+}
+
+func (q *Deploy) GetLastPoppedState() LastPoppedState {
+	return q.lastPopped
+}
+
+func (q *Deploy) SetLastPoppedState(state LastPoppedState) {
+	q.lastPopped = state
+}
```

Comment:
This doesn't feel right to me.  A queue shouldn't have to ever worry about something once it's not in the queue anymore.  Feels like this should be something you query the worker for to get what is in-progress.  The worker already maintains state around whether it's working or waiting and this is very related to that.